ext/socket/raddrinfo.c

```diff
@@ -297,17 +345,24 @@ rb_getaddrinfo(const char *node, const char *service,
     if (ret == 0)
         allocated_by_malloc = 1;
     else {
+        VALUE scheduler = rb_fiber_scheduler_current();
+
+        if (scheduler != Qnil && node && !(hints->ai_flags & AI_NUMERICHOST) &&
+            strncmp(node, "localhost", strlen(node))) {
```

Comment:
There is one interesting concern I have here.

I believe some people will try to use this scheduler hook to do name rewriting. For example, mocking hosts by rewriting DNS.

I wonder if we should therefore avoid making assumptions about the name like `localhost`.